url.bs

```diff
@@ -2476,39 +2476,83 @@ background information. [[!HTML]]
 <h3 id=url-rendering>URL rendering</h3>
 <!-- See https://www.w3.org/Bugs/Public/show_bug.cgi?id=27641 for context -->
 
-<p>A <a for=/>URL</a> should be rendered in its <a lt="URL serializer">serialized</a>
-form, with these modifications:
+<p>A <a for=/>URL</a> should be rendered in its <a lt="URL serializer">serialized</a> form, with
+modifications described below when the primary purpose of displaying a URL is to have the user make
+a security decision (e.g., users are expected to make trust decisions based on a URL rendered in the
+browser address bar).
+
+<h4 id=url-rendering-simplification>Simplify non-human-readable or irrelevant components</h4>
+
+<p>Remove components that may provide opportunities for spoofing or distract from security-relevant
+information:
 
 <ul class=brief>
- <li><p>A <a for=/>URL</a>'s <a for=url>username</a> and <a for=url>password</a> should
- not be rendered as they can be mistaken for a <a for=/>URL</a>'s <a for=url>host</a>.
- E.g., consider <code>https://examplecorp.com@attacker.example/</code>.
+ <li><p>Browsers are encouraged to only render a URL’s <a for=url>host</a> in places where it is
+ important for users to distinguish between the host and other parts of the URL such as the <a
+ for=url>path</a>. Browsers may further consider rendering only the URL’s host's <a
+ for=host>registrable domain</a> to remove spoofing opportunities posed by subdomains (e.g.,
+ <code>https://examplecorp.attacker.com/</code>).
 
- <li><p>A <a for=/>URL</a>'s <a for=url>host</a> should be rendered using
- <a>domain to Unicode</a>.
+ <li><p>A <a for=/>URL</a>'s <a for=url>username</a> and <a for=url>password</a> should not be
+ rendered as they can be mistaken for a <a for=/>URL</a>'s <a for=url>host</a> (as in, e.g.,
+ <code>https://examplecorp.com@attacker.example/</code>).
 
- <li><p>Other parts of the <a for=/>URL</a> should have their sequences of
- <a>percent-encoded bytes</a> replaced with code points resulting from
- <a>percent decoding</a> those sequences converted to bytes, unless that renders those
- sequences invisible.
+ <li><p>A URL can be rendered without its <a for=url>scheme</a> if the display surface only ever
+ permits a single scheme (such as a browser feature that omits <code>https://</code> because it is
+ only enabled for secure origins). Otherwise, the scheme may be replaced or supplemented with a
+ human-readable string (e.g., "Not secure") and/or a security indicator icon.
+
+ <li><p>A leading <code>www</code> domain label can be omitted to simplify the host, except when it
+ is part of the <a for=host>registrable domain</a>.
```

Comment:
This one led to some issues right? It seems this can also be problematic if the site isn't reachable without it.

Reply:
The main issue was a bug in the implementation, which has now been fixed. We are planning on re-landing the change in Chrome and so it would be great to have it documented here.

Chrome's implementation unelides and shows the full domain when you copy/select/edit the URL, which is intended to minimize confusion if the site is not reachable without 'www'. We also did an analysis of top sites and found that it's uncommon to serve different content at www vs bare domain, or have one be reachable but not the other. (Only 2% of Alexa top 500 hosts serve different content at www vs bare domain.)

Finally, I think this guidance is also somewhat in line with what Safari does: they don't elide www specifically, but they show the registrable domain only. The underlying reasoning is the same -- it's okay to show a subset of the actual host in service of helping users understand the site identity better. (In fact, one possibility might be to fold this bullet in with the first bullet and phrase it more as "Browsers are encouraged to render a URL's host only. Browsers may further consider rendering only the most relevant parts of a host, such as omitting a leading 'www' domain label or rendering the registrable domain only.") WDYT?

Reply:
I cannot reproduce Safari only showing the registrable domain. E.g., for https://a.x.testsuite.org/ I see `a.x.testsuite.org`.

Reply:
Ah interesting, I thought they were but it looks like it's more complicated than that... they at least omit `www` (see screenshot on `www.google.com`).

![image](https://user-images.githubusercontent.com/602511/54767377-4418ff80-4bba-11e9-8739-398c7ff50977.png)
![image](https://user-images.githubusercontent.com/602511/54767393-4a0ee080-4bba-11e9-9181-a657ec9c3ce5.png)

Maybe @johnwilander knows the specifics.

Reply:
Seems they omit `m` too:

![image](https://user-images.githubusercontent.com/602511/54767729-09639700-4bbb-11e9-8fcf-1dde47d81bfe.png)
![image](https://user-images.githubusercontent.com/602511/54767735-0e284b00-4bbb-11e9-8723-86324b821826.png)

Reply:
@annevk are you okay leaving this in here given that Safari does it and Chrome is planning too as well?

Reply:
Yeah, I wonder if we should generalize it given the `m` case you discovered.

Reply:
Ok, that seems reasonable. I collapsed it into the first bullet and made it about simplifying the host generally, with omitting `www` or `m` or displaying registrable domain only listed as examples.